apis/v1alpha2/validation/httproute_test.go

```diff
@@ -312,12 +314,105 @@ func TestValidateHTTPRoute(t *testing.T) {
 	}
 	for _, tt := range tests {
 		// copy variable to avoid scope problems with ranges
-		tt := tt
 		t.Run(tt.name, func(t *testing.T) {
-			errs := ValidateHTTPRoute(&tt.hRoute)
+			errs := validateHTTPRouteUniqueFilters(tt.hRoute.Spec.Rules, field.NewPath("spec").Child("rules"))
 			if len(errs) != tt.errCount {
 				t.Errorf("ValidateHTTPRoute() got %v errors, want %v errors", len(errs), tt.errCount)
 			}
 		})
 	}
 }
+
+func TestValidateHTTPBackendUniqueFilters(t *testing.T) {
+	var testService = "testService"
+	var specialService = "specialService"
+	tests := []struct {
+		name     string
+		hRoute   gatewayv1a2.HTTPRoute
```

Comment:
Can the input here just be a `gatewayv1a2.HTTPRouteRule` to save some layers of indentation?

Reply:
56515a744c7cfd05846877db87e7377ff937a48d